packages/scenario-harness/ui/app.js

```
@@ -434,6 +435,96 @@ function connectWebSocket() {
+  for (const { id } of ENGINE_WARNINGS) {
+    const cb = $(id);
+    if (!cb) continue;
+    cb.addEventListener("change", () => {
+      const next = readEngineWarningsMaskFromUi();
+      // Optimistic local update so subsequent reads (this function on a
+      // rapid second click, `readReadbackSnapshot()` on a readback fire)
+      // see the new value immediately. The WS echo arriving later is a
+      // no-op confirmation. Guard against the controller state not being
+      // populated yet on first boot.
+      if (state.controller?.telemetry) state.controller.telemetry.EngineWarnings = next;
+      post("/api/telemetry", { patch: { EngineWarnings: next } }).catch((err) => alert(err.message));
```

ESLint

[error] 'alert' is not defined.

(no-undef)